pipelines/build/common/config_regeneration.groovy

```diff
@@ -0,0 +1,129 @@
+@Library('local-lib@master')
+import common.IndividualBuildConfig
+import groovy.json.*
+
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+* This file is a job that regenerates all of the build configurations in pipelines/build/openjdk*_pipeline.groovy to ensure that race
+* conditions are avoided for concurrent builds.
+*
+* This:
+* 1) Is triggered by a githook from openjdk-build whenever there is a push to the master branch
+* 2) Checks if there are any pipelines in progress. If so, this job will run once they have finished and will force other pipelines to  * await for its completion before executing
+* 3) Regenerates the all the job configurations to so they are all matched and in sync with each other. This will avoid one job
+* configuration overwritting another.
+*/
+
+node ("master") {
+    context.stage("Check") {
+        // Download jenkins helper
+        def JobHelper = context.library(identifier: 'openjdk-jenkins-helper@master').JobHelper
+
+        // Get all pipelines (use jenkins api)
+        def parser = new JsonSlurper()
+        def pipelines = []
+
+        try {
+            def get = new URL("https://ci.adoptopenjdk.net/job/build-scripts/api/json?tree=jobs[name]&pretty=true&depth1").openConnection()
+            def rc = get.getResponseCode()
+            def response = parser.parseText(get.getInputStream().getText())
+
+            // Parse api response
+            response.jobs.name.each{ job -> 
+                if (job.contains("pipeline")) {
+                    pipelines.add(job)
+                }
+            }
+
+        } catch (Exception e) {
+            // Failed to connect to jenkins api or a parsing error occured
+            println "Failure on jenkins api connection or parsing. API response code: ${rc}\nError: ${e.getLocalizedMessage()}"
+            currentBuild.result = 'FAILURE'
+        } 
+        finally {
+            // Query jobIsRunning jenkins helper for each pipeline
+            pipelines.each { pipeline ->
+                def inprogress = true
+
+                while (inprogress) {
+                    println "Checking if ${pipeline} is running..."
+                    if (JobHelper.jobIsRunning(pipeline as String)) {
+                        println "${pipeline} is running. Sleeping..."
```

Comment:
extract `900` to a var and then you can say sleeping for $var.  Actually maybe even more descriptive, add why you are sleeping.  'waiting to queue up next pipeline run` or some such